### ISSUE TITLE
Installation on own server

### ISSUE THREAD

**Issue by tim-krieger** [sgtpep/booksie#4]
Thank you so much for this great app!

Could you please describe a way to install Booksie on our own server?

**Comment by sgtpep** [comment]
Sorry for the late reply. I'm very happy to hear you liked it, although it still has a lot of room for improvement :)

It consists of two repositories:
- https://github.com/sgtpep/booksie — the source code that gets compiled down to static files (html/css/js), which could be served using any web server, like nginx or apache.
- https://github.com/sgtpep/booksie-data — contains 1) scripts that could be run manualy or on schedule that check and download new books and create the index file with a list of them (https://data.booksie.org/books.json); 2) stored books and the index file in gh-pages branch.

Steps on building the web app itself from the first repository (UNIX-like OS, Mac OS or Linux, is assumed):
1. `git clone --depth=1 https://github.com/sgtpep/booksie.git`
2. `cd booksie`
3. `npm i` (Node.js needs to be installed beforehand.)
4. ~You might need to find the mentions of `booksie.org` in source files and replace them, if needed. The build script (`./build`) also has the hardcoded path `https://data.booksie.org/`, from which it will look for the index file and download actual books and their covers.~ `cp config.example.json config.json` and edit `config.json` changing `https://data.bookise.org/` to the URL where PDF files of books and the index file reside. The URL can be relative as well if the data is accessible through the same domain. In this case, the value could be just something like `/books/` instead of a full URL. 
5. `CI=1 ./build`
6. The output static files that could to be served to the outside world will be in `dist` directory.

Let me know, if you're interested in deploying https://github.com/sgtpep/booksie-data on your server, as well, I'll be happy to provide the instructions.

**Comment by tim-krieger** [comment]
Thank you a lot for your detailled answer! Yes, it would be great if you could give us instructions for deploying booksie-data on our server.

**Comment by sgtpep** [comment]
I've updated the 4th step above after I added a way to override the URL to books from https://data.bookise.org/ to any custom location.

The gist of setting up `bookie-data` project is pretty much similar to the above.

1. `git clone --depth=1 https://github.com/sgtpep/booksie-data.git`
2. `cd booksie-data`
3. `npm i`
4. `CI=1 ./build`
5. After downloading pdf files and generating cover images and the index file, the output will be in `dist` directory which could be served using any web server that is able to serve static files, like nginx or Apache.

The instructions above assume that you have some Linux-like server that you're in control of. As an alternative, this project could be hosted using only free services, like Github Pages for serving static files and Travis CI to running all the `./build` steps on schedule, as how it works with bookie.org now.
